WordPress/gutenberg
# Fix image block v6 deprecation

## What?


Fixes an image block deprecation issue surfaced in https://github.com/WordPress/gutenberg/issues/52388#issuecomment-1630025459. I'm not sure if it really solves the original issue because the page should load despite block validation errors.

> Greenshift has a lot of block validation errors:
>
> <img width="1411" alt="Screenshot 2023-07-11 at 12 46 21 pm" src="https://github.com/WordPress/gutenberg/assets/6458278/c78e9a58-3af7-4d0f-85ba-35ad3e3ea4b6">

## Why?


Bug fix

## How?


1. Fix `getBorderClassesAndStyles` import.
2. Fix incorrect comment.
3. Add `attributes` and `supports` to the deprecation.
4. Add integration tests for the deprecation.

## Testing Instructions





Integration tests have been added for the deprecation. They should pass.

To manually test, add a v6 image block to a post via the code editor and save:

```html

<figure class="wp-block-image alignleft size-large is-resized has-custom-border is-style-rounded"><img src="https://s.w.org/images/core/5.8/portrait.jpg" alt="" style="border-radius:100%" width="164" height="164"/></figure>

```

You should no longer see the deprecation warning.

```
Block validation: Block validation failed for `core/image` ({name: 'core/image', icon: {…}, keywords: Array(3), attributes: {…}, providesContext: {…}, …}).

Content generated by `save` function:

<figure class="wp-block-image alignleft size-large is-resized has-custom-border is-style-rounded"><img src="https://s.w.org/images/core/5.8/portrait.jpg" alt="" style="border-radius:100%;width:164px;height:164px" width="164" height="164"/></figure>

Content retrieved from post body:

<figure class="wp-block-image alignleft size-large is-resized has-custom-border is-style-rounded"><img src="https://s.w.org/images/core/5.8/portrait.jpg" alt="" style="border-radius:100%" width="164" height="164"/></figure>
```

### Testing Instructions for Keyboard


## Screenshots or screencast
